pinot-common/src/main/java/com/linkedin/pinot/common/utils/CommonConstants.java

```diff
@@ -252,6 +252,10 @@ public ServerType getServerType() {
     public static final String DEFAULT_COLUMN_MIN_MAX_VALUE_GENERATOR_MODE = "TIME";
   }
 
+  public static class Controller {
+    public static final String PREFIX_OF_CONFIG_OF_SEGMENT_FETCHER_FACTORY = "pinot.controller.segment.fetcher";
```

Comment:
the segment fetcher code is actually used by both controller and server (usage in server code in [here](https://github.com/linkedin/pinot/blob/master/pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentFetcherAndLoader.java#L78)) I am not sure prefixing it with either controller/server is a good way to do. Maybe we should consider the naming something like pinot.common.segment.fetcher?

Reply:
The  config for segment fetcher in controller   should be prefixed by pinot.controller.

the config for segment fetcher in server should be prefixed by pinot.server.

All configs inside have to be duplicated for controller and server as needed depending on the fetchers that they are using. 

I don't see a problem with this. Maybe I don't understand your question. 

Changing things to pinot.common will certainly break anyone who has been using pinot.server. for a long time (and it has been there for years now).

Reply:
got it, I misunderstood the changes you made in server starter section earlier but it makes sense now. Thanks!